unidist/core/backends/mpi/core/shared_store.py

```diff
@@ -0,0 +1,753 @@
+        return self.service_buffer[service_index + self.REFERENCES_NUMBER]
+
+    def get_shared_buffer(self, first_index, last_index):
+        """
+        Get the requested range of shared memory
+
+        Parameters
+        ----------
+        first_index : int
+            Start of the requested range.
+        last_index : int
+            End of the requested range. (excluding)
+
+        Notes
+        -----
+        This function is used to synchronize shared memory between different hosts.
+        """
+        return self.shared_buffer[first_index:last_index]
+
+    def delete_service_info(self, data_id, service_index):
+        """
+        Delete service information for the current data Id.
+
+        Parameters
+        ----------
+        data_id : unidist.core.backends.common.data_id.DataID
+        service_index : int
+            The service buffer index.
+
+        Notes
+        -----
+        This function should be called by the monitor during the cleanup of shared data.
+        """
+        with SharedSignaler(self.win_service):
+            # Read actual value
+            old_worker_id = self.service_buffer[service_index + self.WORKER_ID_INDEX]
+            old_data_id = self.service_buffer[service_index + self.DATA_NUMBER_INDEX]
+            old_first_index = self.service_buffer[service_index + self.FIRST_DATA_INDEX]
+            old_references_number = self.service_buffer[
+                service_index + self.REFERENCES_NUMBER
+            ]
+
+            # check if data_id is correct
+            if self._parse_data_id(data_id) == (old_worker_id, old_data_id):
+                self.service_buffer[service_index + self.WORKER_ID_INDEX] = -1
+                self.service_buffer[service_index + self.DATA_NUMBER_INDEX] = -1
+                self.service_buffer[service_index + self.FIRST_DATA_INDEX] = -1
+                self.service_buffer[service_index + self.REFERENCES_NUMBER] = -1
+                self.logger.debug(
+                    f"Rank {communication.MPIState.get_instance().rank}: Clear {old_data_id}. Service index: {service_index} First index: {old_first_index} References number: {old_references_number}"
+                )
+            else:
+                self.logger.debug(
+                    f"Rank {communication.MPIState.get_instance().rank}: Did not clear {old_data_id}, because there are was written another data_id: Data_ID(rank_{old_worker_id}_id_{old_data_id})"
+                )
+                self.logger.debug(
+                    f"Service index: {service_index} First index: {old_first_index} References number: {old_references_number}"
+                )
+                raise RuntimeError("Unexpected data_id for cleanup shared memory")
+
+    def put(self, data_id, data):
+        """
+        Put data to shared memory
+
+        Parameters
+        ----------
+        data_id : unidist.core.backends.common.data_id.DataID
+        data : object
+            The current data
```

Comment:
```suggestion
            The current data.
```
put dot everywhere at the end of a docstring